src/apps/api/views/profiles.py

```diff
@@ -6,12 +6,37 @@
 from django.http import HttpResponse
 from rest_framework.generics import GenericAPIView, RetrieveAPIView
 from rest_framework import permissions
+from rest_framework.viewsets import ModelViewSet
+from rest_framework.response import Response
+from django.urls import reverse
 
-from api.serializers.profiles import MyProfileSerializer
+from api.permissions import IsUserAdminOrIsSelf
+from api.serializers.profiles import MyProfileSerializer, UserSerializer
 
 User = get_user_model()
 
 
+class UserViewSet(ModelViewSet):
```

Comment:
We're going to move away from using `ModelViewSet` due to some hidden hazards, such as an automatically created `destroy` view. It's best we create our viewsets by specifying explicitly the views that we want in order to avoid this pitfall.

Note the definition of `ModelViewSet`:
``` python
from rest_framework import mixins, generics

class ModelViewSet(mixins.CreateModelMixin,
                   mixins.RetrieveModelMixin,
                   mixins.UpdateModelMixin,
                   mixins.DestroyModelMixin,
                   mixins.ListModelMixin,
                   generics.GenericViewSet):
    """
    A viewset that provides default `create()`, `retrieve()`, `update()`,
    `partial_update()`, `destroy()` and `list()` actions.
    """
    pass

```

When writing a new `ViewSet`, pick from this list of mixins and omit the ones that you don't want to be automatically created.